ssl/tls13_enc.c

```diff
@@ -29,14 +36,15 @@ int tls13_hkdf_expand(SSL *s, const EVP_MD *md, const unsigned char *secret,
                              const unsigned char *data, size_t datalen,
                              unsigned char *out, size_t outlen)
 {
-    const unsigned char label_prefix[] = "tls13 ";
+    static const unsigned char label_prefix[] = "tls13 ";
     EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_HKDF, NULL);
     int ret;
     size_t hkdflabellen;
     size_t hashlen;
     /*
-     * 2 bytes for length of whole HkdfLabel + 1 byte for length of combined
-     * prefix and label + bytes for the label itself + bytes for the hash
+     * 2 bytes for length of derived secret + 1 byte for length of combined
+     * prefix and label + bytes for the label itself + 1 byte length of hash
+     * + bytes for the hash itself
```

Comment:
Don't we need to actually add another ```sizeof(uint8_t)``` to the addition, for the length of the hash output?

Reply:
could be more clear this way, but then we should subtract one from sizeof(prefix)

Reply:
So initially I had this:
```
sizeof(uint16_t) + sizeof(uint8_t) +
 + (sizeof(label_prefix) - 1) + TLS13_MAX_LABEL_LEN
 + sizeof(uint8_t) + EVP_MAX_MD_SIZE```
... but dropped it again, since the result is effectively the same.